crates/rattler_networking/src/lib.rs

```diff
@@ -0,0 +1,409 @@
+        let entry = Entry::new(&self.store_key, host)?;
+        match authentication {
+            Authentication::BearerToken(token) => {
+                let password = format!("Bearer {}", token);
+                entry.set_password(&password)
+            }
+            Authentication::Basic { username, password } => {
+                let password = format!("Basic {}:{}", username, password);
+                entry.set_password(&password)
+            }
+            Authentication::CondaToken(token) => {
+                let password = format!("CondaToken {}", token);
+                entry.set_password(&password)
+            }
+        }
+    }
+
+    /// Retrieve the authentication information for the given host
+    pub fn get(&self, host: &str) -> Result<Option<Authentication>, AuthenticationStorageError> {
+        if let Some(cached) = self.authentication_cache.get(host) {
+            return Ok(cached.clone());
+        }
+
+        let entry = Entry::new(&self.store_key, host)?;
+        let password = entry.get_password();
+
+        match password {
+            Ok(password) => Ok(Some(Authentication::from_str(&password).map_err(|_| {
+                AuthenticationStorageError::ParseCredentialsError {
+                    host: host.to_string(),
+                }
+            })?)),
+            Err(keyring::Error::NoEntry) => Ok(None),
+            Err(e) => Err(AuthenticationStorageError::StorageError(e)),
+        }
+    }
+
+    /// Retrieve the authentication information for the given URL
+    pub fn get_by_url<U: IntoUrl>(
+        &self,
+        url: U,
+    ) -> Result<(Url, Option<Authentication>), reqwest::Error> {
+        let url = url.into_url()?;
+
+        if let Some(host) = url.host_str() {
+            let credentials = self.get(host);
+            match credentials {
+                Ok(None) => Ok((url, None)),
+                Ok(Some(credentials)) => Ok((url, Some(credentials))),
+                Err(e) => {
+                    tracing::warn!("Error retrieving credentials for {}: {}", host, e);
+                    Ok((url, None))
+                }
+            }
+        } else {
+            Ok((url, None))
+        }
+    }
+
+    /// Delete the authentication information for the given host
+    pub fn delete(&self, host: &str) -> keyring::Result<()> {
+        let entry = Entry::new(&self.store_key, host)?;
+        entry.delete_password()
+    }
+}
+
+/// A client that can be used to make authenticated requests, based on the [`reqwest::Client`]
+#[derive(Clone)]
+pub struct AuthenticatedClient {
+    /// The underlying client
+    client: Client,
+
+    /// The authentication storage
+    auth_storage: AuthenticationStorage,
+}
+
+impl Default for AuthenticatedClient {
+    fn default() -> Self {
+        AuthenticatedClient {
+            client: Client::default(),
+            auth_storage: AuthenticationStorage::new("rattler"),
```

Comment:
I think the `Default` implementation should be "unauthenticated". Using "rattler" as a default key means you are accessing a giant global variable that is used by everything that is using rattler which seems like a huge red-flag to me.